core/tests/protractor_desktop/learnerDashboard.js

```diff
@@ -153,19 +204,239 @@ describe('Learner dashboard functionality', function() {
+        await elem.setValue(topicId);
+      });
+    await topicsAndSkillsDashboardPage.get();
+    (
+      await
+      topicsAndSkillsDashboardPage.createSkillWithDescriptionAndExplanation(
+        'Skill Learner Dashboard1', 'Concept card explanation', false));
+    await topicsAndSkillsDashboardPage.get();
+    await topicsAndSkillsDashboardPage.navigateToSkillsTab();
+    await topicsAndSkillsDashboardPage.assignSkillToTopic(
+      'Skill Learner Dashboard1', 'Topic 1');
+    await topicsAndSkillsDashboardPage.get();
+    await topicsAndSkillsDashboardPage.navigateToTopicWithIndex(0);
+    await topicEditorPage.addSubtopic(
+      'Subtopic 1', 'subtopic-one', '../data/test2_svg.svg',
+      'Subtopic content');
+    await topicEditorPage.saveTopic('Added subtopic.');
+
+    await topicEditorPage.navigateToTopicEditorTab();
+    await topicEditorPage.navigateToReassignModal();
+
+    await topicEditorPage.dragSkillToSubtopic('Skill Learner Dashboard1', 0);
+    await topicEditorPage.saveRearrangedSkills();
+    await topicEditorPage.saveTopic('Added skill to subtopic.');
+
+    await topicEditorPage.updateMetaTagContent('meta tag content');
+    await topicEditorPage.updatePageTitleFragment('fragment');
+    await topicEditorPage.saveTopic('Added meta tag and page title fragment.');
+
+    await topicEditorPage.publishTopic();
+    /**  There is one topic on the server named Topic 1 which is linked to a
+     * subtopic named Subtopic 1 and a story called Story Title.
+     * Subtopic 1 has one skill in it named Skill Learner Dashboard1.
+     */
+    await learnerDashboardPage.get();
+    await learnerDashboardPage.navigateToHomeSection();
+    await learnerDashboardPage.expectNumberOfTopicsInSuggestedForYou(1);
+    await learnerDashboardPage.expectNumberOfTopicsInContinueWhereYouLeftOff(0);
+    await learnerDashboardPage.navigateToGoalsSection();
+    await learnerDashboardPage.expectNameOfTopicInEditGoalsToMatch(
+      TOPIC_NAME);
+    await learnerDashboardPage.addTopicToLearnerGoals();
+    await learnerDashboardPage.navigateToGoalsSection();
+    await learnerDashboardPage.expectNameOfTopicInCurrentGoalsToMatch(
+      `Learn ${TOPIC_NAME}`);
+    await learnerDashboardPage.navigateToHomeSection();
+    await learnerDashboardPage.expectNumberOfTopicsInContinueWhereYouLeftOff(1);
+    await users.logout();
+  });
+
+  it('should display all the topics that are partially learnt or learnt ' +
+    'in skill proficiency section, learnt topics in completed goals section ' +
+    'and completed stories in completed stories section', async function() {
+    await users.createAndLoginCurriculumAdminUser(
+      'creator@learnerDashboard2.com', 'learnerDashboard2');
+    await createDummyExplorations();
+    var handle = await browser.getWindowHandle();
+    await topicsAndSkillsDashboardPage.get();
+    await topicsAndSkillsDashboardPage.createTopic(
+      'Topic 2', 'topic-two', 'Description', false);
```

Comment:
Please name this topic to be reliably unique

Reply:
Done.